### DISCUSSION TITLE
Images not working on remote

### DISCUSSION THREAD

**Discussion by promise** [community/community#23630] [New to GitHub]
<p>I have images on my local computer and I’m able to open them just fine there. Then, when I push them to the repository, the images gets corrupted and I’m not able to open them anymore. And when I try and clone the repository to another folder, I can’t seem to open the images there either.</p>
<p>Here’s the folder on GH: <a href="https://github.com/biaw/test-area/tree/master/assets/emojis" class="inline-onebox" rel="noopener nofollow ugc">test-area/assets/emojis at master · biaw/test-area · GitHub</a><br>
I’ve uploaded them on Imgur: <a href="https://imgur.com/a/jvQyJv3" class="inline-onebox" rel="noopener nofollow ugc">Imgur: The magic of the Internet</a> - but I’m unsure if Imgur messes with the raw files so I’ve zipped the images as well (link is in the Imgur post since I can’t upload more than two links in a post)</p>
<p>Thanks in advance.<br>
Promise</p>

**Comment by jfullstackdev** [comment]
<p>I also checked that, it’s really corrupted, how come?</p>
<p>if you can’t correctly push that, directly drag and drop the folder on GitHub Web, it will be uploaded too,</p>
<p><img src="https://user-images.githubusercontent.com/47092464/181078369-45a34908-5859-4614-b213-db9fe73b9d58.gif" alt="a888b59e-d004-11e5-80a6-e9f32f17c4d5" data-base62-sha1="pVUQyhkMOjrw4hfZ31LjRqsWArL" width="690" height="416" class="animated"></p>

**Comment by promise** [comment]
<p>That worked, thanks. Seems like one byte was left out…?<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/e/7/e7551ebf3d915ba1b9cad7b6ef93aa5576b8b08f.png" data-download-href="/uploads/short-url/x0sFcWPdzDkU6VS4l5e8Q9hs3ef.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/10573728/181078370-a9839ffd-326d-4022-8710-fcbfc6c23d3c.png" alt="image" data-base62-sha1="x0sFcWPdzDkU6VS4l5e8Q9hs3ef" width="353" height="500" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/e/7/e7551ebf3d915ba1b9cad7b6ef93aa5576b8b08f_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">358×506 22.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>At least it works now, thank you!</p>
<p>Should I keep this open for staff to investigate why pushing doesn’t work or should I just close this?</p>

**Comment by jfullstackdev** [comment]
<p>I don’t know about the specific problem, but it’s up to you to pursue that,</p>

**Comment by promise** [comment]
<p>Seems like it didn’t fix the problem entirely. My local repository now says it has changes to the images and they appear corrupt as well. Discarding the changes doesn’t do anything, and it also happens after re-cloning the remote…</p>
<p><img src="https://user-images.githubusercontent.com/10573728/181078373-802453f8-e6fe-4edf-9d3e-5a3f5ab6e509.gif" alt="gif" data-base62-sha1="pGmh61IUc5oBklb8qPkRVGGrCUy" width="690" height="393" class="animated"></p>
<p>(unselected your reply as the solution so it might get more attention)</p>

**Comment by jfullstackdev** [comment]
<p>it’s not corrupted on GitHub anymore,</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/e/0/e0c918601fc57237e967675ac77e06956390276d.png" data-download-href="/uploads/short-url/w4xNt6OJwDHF2LjtKrq4mIWb2cR.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/47092464/181078376-d415e275-fc6c-47a2-add3-1a1abab17095.png" alt="image" data-base62-sha1="w4xNt6OJwDHF2LjtKrq4mIWb2cR" width="690" height="388" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/e/0/e0c918601fc57237e967675ac77e06956390276d_2_690x388.png, https://aws1.discourse-cdn.com/github/optimized/3X/e/0/e0c918601fc57237e967675ac77e06956390276d_2_1035x582.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/e/0/e0c918601fc57237e967675ac77e06956390276d_2_1380x776.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/e/0/e0c918601fc57237e967675ac77e06956390276d_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1920×1080 126 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/8/0/809d27f725a5ee6ea1535bcf9b1e48c47b92ae1a.png" data-download-href="/uploads/short-url/ilLOWo0KUQkpK0IUM5KLBVjtjZw.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/47092464/181078378-81993de6-023a-46dd-a33d-797be8eb1f20.png" alt="image" data-base62-sha1="ilLOWo0KUQkpK0IUM5KLBVjtjZw" width="690" height="388" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/8/0/809d27f725a5ee6ea1535bcf9b1e48c47b92ae1a_2_690x388.png, https://aws1.discourse-cdn.com/github/optimized/3X/8/0/809d27f725a5ee6ea1535bcf9b1e48c47b92ae1a_2_1035x582.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/8/0/809d27f725a5ee6ea1535bcf9b1e48c47b92ae1a_2_1380x776.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/8/0/809d27f725a5ee6ea1535bcf9b1e48c47b92ae1a_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1920×1080 127 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by promise** [comment]
<p>yeah the images are fine on GitHub, but they’re corrupt when I clone the repo.</p>
<p>I also have a GH action to auto-publish images to Docker, but these images also seem to have corrupt images. Building a docker image on my computer works just fine, so it seems to be a problem with GitHub when cloning.</p>

**Comment by airtower-luna** [comment]
<p>Do you have the repository configured to mangle files in any way on checkout? For example it’s common to configure Git on Windows to adjust line endings in text files. Hypothetically, if there was something off about the configuration and something like that was applied to the images it could mess with a few bytes here and there.</p>

**Comment by promise** [comment]
<p>Not that I know of. The only thing I have is a <code>.gitattributes</code> file but that only enforces CRLF line endings on text files (I think)</p>
<pre><code class="lang-auto">* text eol=crlf
</code></pre>

**Comment by jfullstackdev** [comment]
<p>back in the days, when I was learning network programming in Java, the line ending is really that critical, the programmer must really be knowledgeable about that,</p>
<p>so yup, it’s that critical, even here</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="promise" data-post="9" data-topic="218374">
<div class="title">
<div class="quote-controls"></div>
 promise:</div>
<blockquote>
<p><code>* text eol=crlf</code></p>
</blockquote>
</aside>
<p>The pattern <code>*</code> matches <em>every</em> path, so this applies text mode and eol adjustments for all files. That means it should actually affect the images, too.</p>
<p>You can uncomment that line to see if that helps with the images, and if it does adjust the pattern to only affect actual text files (or add an exemption for <code>*.png</code>). :smiley_cat:</p>

**Comment by promise** [comment]
<p>Thanks guys, after a lot of troubleshooting, it finally worked. I set the assets-folder to binary :+1:</p>
